Model/Indexer/Provider/ChangeLogProvider.php

```diff
@@ -34,27 +34,38 @@
  *
  */
 
-namespace Nosto\Tagging\Util;
+namespace Nosto\Tagging\Model\Indexer\Provider;
 
-use Magento\Framework\App\ObjectManager;
 use Magento\Framework\App\ResourceConnection;
 
-class IndexerChangeLog
+class ChangeLogProvider
 {
+
+    /** @var ResourceConnection */
+    private $resourceConnection;
+
+    /**
+     * ChangeLogProvider constructor.
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(ResourceConnection $resourceConnection)
+    {
+        $this->resourceConnection = $resourceConnection;
+    }
+
     /**
-     * @return array
+     * @param string $tableName
+     * @return null
      */
-    public static function countInvalidateCLRows()
+    public function countCLTableEntries(string $indexerName)
     {
-        $objectManager = ObjectManager::getInstance(); // Instance of object manager
-        /** @var ResourceConnection $resource */
-        $resource = $objectManager->get('Magento\Framework\App\ResourceConnection');
-        $connection = $resource->getConnection();
-        $sql = 'SELECT count(*) AS total FROM nosto_index_product_invalidate_cl';
-        $result = $connection->fetchAll($sql);
-        if (is_array($result) && count($result)>0){
-            return $result[0];
+        $connection = $this->resourceConnection->getConnection();
```

Comment:
This looks a bit off with plain SQL queries. Check the Mview module: https://github.com/magento/magento2/tree/2.3/lib/internal/Magento/Framework/Mview. I think most of the stuff we need has been implemented already in that module.